### PR TITLE
[INLONG-11044][SDK] Transform SQL supports string judgment functions(IS_ALPHA、IS_DECIMAL、IS_DIGIT)

### LINKED ISSUE #11044: [Feature][SDK] Transform SQL supports string judgment functions (IS_ALPHA、IS_DECIMAL、IS_DIGIT)
### Description

- parent issue #10796 
- FlinkSql Official Description: https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/functions/systemfunctions/

IS_ALPHA(string): Returns true if all characters in string are letter, otherwise false.
IS_DECIMAL(string): Returns true if string can be parsed to a valid numeric, otherwise false.
IS_DIGIT(string): Returns true if all characters in string are digit, otherwise false.

### Use case

```
IS_ALPHA('a1') -> false
IS_ALPHA('a')  -> true
IS_ALPHA(null)  -> false

IS_DECIMAL('a1') -> false
IS_DECIMAL('1')  -> true
IS_DECIMAL('1.3')  -> true
IS_DECIMAL(null)  -> false

IS_DIGIT('a1') -> false
IS_DIGIT('1')  -> true
IS_DIGIT('1.3')  -> false
IS_DIGIT(null)  -> false

```

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zkplo**
Fixes #11044 

### Motivation

Add three function classes: IsAlphaFunction, IsDecimalFunction, IsDigitFunction. Also, add the corresponding unit test classes: TestIsAlphaFunction, TestIsDecimalFunction, TestIsDigitFunction.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsAlphaFunction.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObject = stringParser.parse(sourceData, rowIndex, context);
+        if (stringObject == null) {
+            return false;
+        }
+        String string = OperatorTools.parseString(stringObject).toLowerCase();
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsAlphaFunction.java:48]: It is extra string object.
It is better for ((chr >='a' && chr <='z') || (chr >='A' && chr <='Z'))

Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsAlphaFunction.java]: Thanks. done.

**Review by luchunliang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsDecimalFunction.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        Object stringObject = stringParser.parse(sourceData, rowIndex, context);
+        if (stringObject == null) {
+            return false;
+        }
+        String string = OperatorTools.parseString(stringObject).toLowerCase();
+        if (string.isEmpty()) {
+            return false;
+        }
+        try {
+            new BigDecimal(string);
```
Comment (vernedeng) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsDecimalFunction.java:55]: It's more elegant that use Regular Expressions to check big decimal
https://www.cnblogs.com/cstdio1/p/12128841.html

Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/IsDecimalFunction.java]: Great! Done.

**Review by luchunliang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**